File: uStock-Development-Android/taskThree_models/Data Structures/AIModels/AIChat.swift
//
//  AIChat.swift
//  uStockIOS-V2-AI
//
//  Created by Gal Cohavy on 6/27/23.
//

import Foundation

public struct AIChat: Codable {
    public let id: String
    let creator: String
    let assistantName: String
    let messages: [Message]
    let createdAt: Date
    let updatedAt: Date
    
    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case creator
        case assistantName
        case messages
        case createdAt
        case updatedAt
    }
}

File: uStock-Development-Android/app/src/main/java/data_structures/ai_models/AIChat.kt
package data_structures.ai_models
import data_structures.DateSerializer
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.util.Date

@Serializable
data class AIChat(
    @SerialName("_id") val id: String,
    val creator: String,
    val assistantName: String,
    val messages: List<Message>,
    @Serializable(with = DateSerializer::class)
    val createdAt: Date,
    @Serializable(with = DateSerializer::class)
    val updatedAt: Date
)
